### DISCUSSION TITLE
Full TypeScript definitions

### DISCUSSION THREAD

**Discussion by letmaik** [vueform/vueform#208] [Ideas]
The TypeScript definitions are mostly resolving to `any` at the moment. Is there any plan to have more complete definitions? Maybe even gradually adopting TypeScript for vueform's source itself and then getting types out of the box?

**Comment by adamberecz** [comment]
Yes, the type definitions are currently really only there to give users prop name hints so they don't have to look up everything in the docs. It doesn't add proper type safety - I'm aware. At this point there's probably nothing that I'd love to do more than rewriting Vueform in TypeScript. Unfortunately that's something I don't see doing anytime soon, as I'm currently maintaining the whole ecosystem alone while also building a business.

The only way I see this happening in the near future if someone takes on writing a proper type system in DefinitelyTyped or by becoming a co-maintainer.

**Reply by Zebnastien** [reply]
Zod has the hability to extract form type which is a killer feature for server side validation.
I would love to see this coming to Vueform along with TypeScript support.

By the way, your project is awesome and very well made.
I wish you best luck in your business!
